### DISCUSSION TITLE
DELETED

### DISCUSSION THREAD

**Discussion by ziqbal** [PyMySQL/mysqlclient#653] [Trash]

**Comment by methane** [comment]
Did you use venv?
Do not install packages from PyPI into system Python directly.

**Reply by methane** [reply]
You can avoid venv if and only if you are using Python not managed by system package manager.
For example, python:3.11 docker doesn't require venv because it uses non-system Python.
But you are using ubuntu image and system Python. You must use venv even in container.

**Reply by methane** [reply]
Again, if you don't want to use venv in contianer, you must use Python outside of system package manager like apt. For example, you can skip venv and directly install 3rd party modules from PyPI with `python:3.11` image.

On the other hand, your pip and setuptools are too old when you are using ubuntu:22.04 system python. 

They don't support PEP 517 so you will see more and more build error when installing from source packages.
Now most source packages are PEP 517 based, because recent pip shows warning for old source packages doesn't use PEP 517.

> and make a mental note to switch to an alternative python mysql library.

Such a threatening words doesn't help OSS world.

Note that you are saying "I can not install shine new package from open package registry with 1.5 years old unsupported legacy/deprecated installer". It doesn't make sense at all.

**Comment by ziqbal** [comment]


**Reply by methane** [reply]
USE venv unless you are using "python" docker image.
Your "Using venv within a container seems an overkill." is totally wrong idea.
That's all. mysqlclient supports ubuntu docker image.
